doc/mgmt/Sonic_Config_Mgmt_HLD.md

```diff
@@ -0,0 +1,187 @@
+### 3.2.1 CONFIG DB
+### 3.2.2 APP DB
+### 3.2.3 STATE DB
+### 3.2.4 ASIC DB
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+### 3.3.2 Other Process
+Not Applicable
+
+## 3.4 SyncD
+Not Applicable
+
+## 3.5 SAI
+Not Applicable
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+
+A new sonic yang (sonic-config-mgmt.yang) provides the interface for configuration and status.
+
+```
+module: sonic-config-mgmt
+
+  rpcs:
+    +---x copy
+       +---w input
+       |  +---w source?        string
+       |  +---w overwrite?     boolean
+       |  +---w destination?   string
+       +--ro output
+          +--ro status?   string
+
+```
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands
+As described in section 1.1.2
+
+#### 3.6.2.2 Show Commands
+Not Applicable
+
+#### 3.6.2.3 Debug Commands
+Not Applicable
+
+#### 3.6.2.4 IS-CLI Compliance
+The following table maps SONIC CLI commands to corresponding IS-CLI commands. The compliance column identifies how the command comply to the IS-CLI syntax:
+
+- **IS-CLI drop-in replace**  – meaning that it follows exactly the format of a pre-existing IS-CLI command.
+- **IS-CLI-like**  – meaning that the exact format of the IS-CLI command could not be followed, but the command is similar to other commands for IS-CLI (e.g. IS-CLI may not offer the exact option, but the command can be positioned is a similar manner as others for the related feature).
+- **SONIC** - meaning that no IS-CLI-like command could be found, so the command is derived specifically for SONIC.
+
+|       CLI Command       | Compliance   | click CLI                    | Deviation
+|:-----------------------:|:-------------|------------------------------|---------------
+| write memory            | IS-CLI-like  |  config save                  |
+|  copy <*filename*> running-configuration [overwrite]    |    IS-CLI-like          |  config load <*filename*><br> config reload <*filename*>              | This incorporates copy and restart of services.
+| copy running-configuration <*filename*\>  |     IS-CLI-like         |  config save <*filename*>                 |
+|copy startup-configuration  running-configuration  | IS-CLI-like   | config  load <br> config reload |   |   |   |
+
+**Deviations from IS-CLI:** If there is a deviation from IS-CLI, Please state the reason(s).
+
+
+### 3.6.3 REST API Support
+Rest API is supported through the sonic-config-mgmt.yang.
+
+# 4 Flow Diagrams
+Not applicable.
+
+# 5 Error Handling
+Not applicable.
+
+# 6 Serviceability and Debug
+Not applicable.
+
+# 7 Warm Boot Support
+Not applicable.
+
+# 8 Scalability
+Not applicable.
+
+# 9 Unit Test
+List unit test cases added for this feature including warm boot.
+CLI test cases
+1. Execute 'write memory'. Default path applied. Verify config_db.json file.
+2. Execute 'copy \<filename\> running-configuration [reload]'. Verify config flush. New config loaded from *filename* into redis:configDB with reload.
```

Comment:
Shouldn't this be overwrite instead of reload

copy \<filename\> running-configuration [**overwrite**]'

Reply:
HLD updated.